iron-core/src/main/java/io/axway/iron/core/internal/StoreManagerFactoryBuilderConfigurator.java

```diff
@@ -0,0 +1,201 @@
+    private static final Pattern VARIABLE_REGEX_PATTERN = Pattern.compile("\\$\\{(.*?)\\}");
+
+    public void fill(StoreManagerFactoryBuilder storeManagerFactoryBuilder, Properties properties) {
+        Map<Type, BuilderImplConfig> componentBuilders = findComponentBuilders(properties);
+        for (Map.Entry<Type, BuilderImplConfig> entry : componentBuilders.entrySet()) {
+            BuilderImplConfig config = entry.getValue();
+            Supplier<?> builder = instantiateBuilder(config.supplierClass(), properties, config.baseName());
+            buildAndAssign(builder, entry.getKey(), storeManagerFactoryBuilder);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<Type, BuilderImplConfig> findComponentBuilders(Properties properties) {
+        Map<Type, BuilderImplConfig> map = new HashMap<>();  // ComponentBuilder -> property name base
+        for (String propertyName : properties.stringPropertyNames()) {
+            String className = getProperty(properties, propertyName);
+            if (className != null) {
+                try {
+                    Class clazz = Class.forName(className);
+                    if (Supplier.class.isAssignableFrom(clazz)) {
+                        Type type = ((ParameterizedTypeImpl) clazz.getGenericInterfaces()[0]).getActualTypeArguments()[0];
+                        if (map.containsKey(type)) {
+                            LOG.warn("A {} is already configured (with {})", type.getTypeName(), map.get(type).supplierClass().getName());
+                        }
+                        BuilderImplConfig builderImplConfig = new BuilderImplConfig(clazz, propertyName);
+                        LOG.info("Adding the configured class in the builder list: {{}: {}}", type.getTypeName(), className);
+                        map.put(type, builderImplConfig);
+                    }
+                } catch (ClassNotFoundException e) {
+                    // nothing to do
+                }
+            }
+        }
+        return map;
+    }
+
+    private Supplier<?> instantiateBuilder(Class<Supplier<?>> componentBuilderClazz, Properties properties, String baseName) {
+        try {
+            Constructor<Supplier<?>> constructor = componentBuilderClazz.getConstructor();
+            Supplier<?> builder = constructor.newInstance();
+
+            for (Method method : componentBuilderClazz.getDeclaredMethods()) {
+                if (method.getName().startsWith("set")) {
+                    Class<?>[] parameterTypes = method.getParameterTypes();
+                    if (parameterTypes.length == 1) {
+                        String propertyName = method.getName().substring(3);
+                        Object value = getPropertyValue(parameterTypes[0], properties, baseName, propertyName);
+                        if (value != null) { // do not call setter if value is null
+                            method.invoke(builder, value);
```

Comment:
so we can't set the FileStoreFactoryBuilder "limitedSize" to null with that ?

Reply:
Good catch!
Now, a key without a value is considered as a set(null), and an absent key will keep the defaut value (if present)